src/test/e2e/api/admin/api-token.auth.e2e.test.ts

```diff
@@ -175,33 +176,32 @@ test('Token-admin should be allowed to create token', async () => {
 test('A role with only CREATE_PROJECT_API_TOKEN can create project tokens', async () => {
     expect.assertions(0);
 
-    const preHook = (app, config, { userService, accessService }) => {
+    const preHook = (
+        app,
+        config,
+        {
+            userService,
+            accessService,
+        }: { userService: UserService; accessService: AccessService },
+    ) => {
         app.use('/api/admin/', async (req, res, next) => {
-            const role = await accessService.getRootRole(RoleName.VIEWER);
+            const role = (await accessService.getRootRole(RoleName.VIEWER))!;
             const user = await userService.createUser({
                 email: 'powerpuffgirls_viewer@example.com',
                 rootRole: role.id,
             });
-            req.user = user;
             const createClientApiTokenRole = await accessService.createRole({
                 name: 'project_client_token_creator',
                 description: 'Can create client tokens',
-                permissions: [],
+                permissions: [{ name: CREATE_PROJECT_API_TOKEN }],
                 type: 'root-custom',
             });
-            await accessService.addPermissionToRole(
-                role.id,
-                CREATE_PROJECT_API_TOKEN,
-            );
             await accessService.addUserToRole(
                 user.id,
                 createClientApiTokenRole.id,
                 'default',
             );
-            req.user = await userService.createUser({
-                email: 'someguyinplaces@example.com',
-                rootRole: role.id,
-            });
```

Comment:
I've refactored this method, not sure what we were trying to do here:
1. **Get Viewer built-in role**
2. Create a user A
3. Create a new role "token creator" without permissions
4. **Add CREATE_PROJECT permission to the Viewer role (shouldn't be possible to add permissions programmatically)**
5. Add user A to the "token creator role" in project 'default' (now A has the root role Viewer and "token creator role" in 'default')
6. **Create a new user B with root role Viewer (that now has the new permission)**
7. **Associate req.user to the newly created user B**

Then the test tries to create an API token using the newly created user B.

Why did we take all those efforts to setup user A if we're not using it in the test? Apparently the only important steps are the ones in **bold**. At least taking into account the title of the test: `A role with only CREATE_PROJECT_API_TOKEN can create project tokens`

Reply:
Guessing copy paste bug. Something something don't trust tests you haven't seen fail